webapp/packages/core-ui/src/Form/FormState.ts

```diff
@@ -90,42 +86,36 @@ export class FormState<TState> implements IFormState<TState> {
       mode: observable,
       parts: observable.ref,
       promise: observable.ref,
-      exception: observable.ref,
-      isSaving: observable.ref,
       state: observable,
+      isSaving: computed,
+      exception: computed,
       isDisabled: computed,
       setMode: action,
       setPartsState: action,
-      setException: action,
       setState: action,
+      isChanged: computed,
+      isError: computed,
+      isCancelled: computed,
     });
   }
 
-  isLoading(): boolean {
-    return this.promise !== null || this.dataContext.get(DATA_CONTEXT_LOADABLE_STATE)!.loaders.some(loader => loader.isLoading());
-  }
-
-  isLoaded(): boolean {
-    if (this.promise) {
-      return false;
-    }
-    return this.dataContext.get(DATA_CONTEXT_LOADABLE_STATE)!.loaders.every(loader => loader.isLoaded());
+  get exception(): Error | (Error | null)[] | null {
+    return Array.from(this.parts.values())
+      .map(part => part?.exception)
+      .flat()
+      .filter(isNotNullDefined);
   }
 
-  isError(): boolean {
-    return this.dataContext.get(DATA_CONTEXT_LOADABLE_STATE)!.loaders.some(loader => loader.isError());
+  get isError(): boolean {
+    return Array.from(this.parts.values()).some(part => part.isError());
   }
```

Comment:
for optimization you better to create private computed getter for part's value `return Array.from(this.parts.values())` and reuse it in other computeds, but don't forget to configure proper `equals` method

example:
```
parts: computed<type[]>({
        equals: isArraysEqual,
      }),
```